src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -57,27 +70,144 @@ def execute(self):
+                msg = "\n----------------------------------------------------------"
+                msg += "\nMSUnmergedRSE: %s"
+                msg += "\n----------------------------------------------------------"
+                self.logger.debug(msg, pformat(rse))
+                totalNumRses += 1
+                if self._checkClean(rse):
+                    numRsesCleaned += 1
+                totalNumFiles += rse['counters']['totalNumFiles']
+                numFilesDeleted += rse['counters']['numFilesDeleted']
+            except Exception as ex:
+                msg = "%s: General error from pipeline. RSE: %s. Error:  \n%s. "
+                msg += "\nWill retry again in the next cycle."
+                self.logger.exception(msg, self.plineUnmerged.name, rse['rse'], str(ex))
+                continue
+        return totalNumRses, totalNumFiles, numRsesCleaned, numFilesDeleted
+
+    def filterRseList(self, rseList):
+        """
+        Applies any additional constraints about the subset of RSEs on which
+        the current service instance is about to work
+        :param rseList: Initial list of RSEs from which the final set should be build
+        :return:        A list of RSEs
+        """
+        return rseList
+
+    def cleanFiles(self, rse):
+        """
+        The method to implement the actual deletion of files for an RSE.
+        :param rse: MSUnmergedRSE object to be cleaned
+        :return:    The MSUnmergedRSE object
+        """
+        try:
+            # for fileUnmerged in rse['files']['toDelete']:
+            #     try:
+            #         self.gfalCommand(rse['delInterface'], fileUnmerged)
+            #         rse['counters']['numFilesDeleted'] += 1
+            #         rse['files']['deletedSuccess'].append(fileUnmerged)
+            #     except Exception as ex:
+            #         rse['files']['deletedFail'].append(fileUnmerged)
+            #         msg = "Error while trying to delete file: %s for RSE: %s"
+            #         msg += "Will retry in the next cycle. Err: %s"
+            #         self.logger.debug(msg, fileUnmerged, rse['name'], str(ex))
+            rse['isClean'] = self._checkClean(rse)
+        except Exception as ex:
+            msg = "Error while cleaning RSE: %s"
+            msg += "Will retry in the next cycle. Err: %s"
+            self.logger.debug(msg, rse['name'], str(ex))
+
+        return rse
+
+    def _checkClean(self, rse):
+        """
+        A simple function to check if every file in an RSE's unmerged area have
+        been deleted
+        :param rse: The RSE to be checked
+        :return:    Bool: True if all files found have been deleted, False otherwise
+        """
+        return rse['counters']['totalNumFiles'] == rse['counters']['numFilesDeleted']
+
+    def resetCounters(self):
+        """
+        A simple function for zeroing the deleted files counters.
+        """
+
+        for rse in self.rseCounters:
```

Comment:
Not exactly... this function was intended to not only clears the counters, but it should also initializes them if not existent. As you can see the very top in the `__init__` those counters are empty dictionaries. But the place this method is called is not the best we do not have the `rseList` yet ... etc. We need to have a quick discussion offline .. on the strategy how to use those, because they are actually duplicates of the already existing counters which are passed between function calls, but I like keeping them in the object more BTW.

Reply:
Oh, I see. You're correct!